probe/websocket/ws_test.go

```diff
@@ -50,6 +51,13 @@ func TestWSPing(t *testing.T) {
 		}
 	}()
 
+	// wait for http server to start
+	conn, err := net.DialTimeout("tcp", "localhost:18188", 30*time.Second)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Consider handling the error returned by `net.DialTimeout` in a more informative way. Instead of using `log.Fatal(err)`, which will stop the execution of the program, you could fail the test with a descriptive message. This would make it easier to understand the cause of the test failure if the connection cannot be established. [important]